### ISSUE TITLE
Datepicker violently crashes when users select a non-gregorian calendar

### ISSUE THREAD

**Issue by sbycrosz** [henninghall/react-native-date-picker#344]
**Describe the bug**

![Simulator Screen Shot - iPhone 11 Pro Max - 2021-06-16 at 15 41 12](https://user-images.githubusercontent.com/4867178/122179149-d1db1680-ceb9-11eb-9862-4e5f9d1ca2bb.png)

```
Exception thrown while executing UI block: *** -[__NSArrayM objectAtIndex:]: index 9223372036854775807 beyond bounds [0 .. 1]

__44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke
    RCTUIManager.m:1183
__44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke.489
__RCTExecuteOnMainQueue_block_invoke
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_callback_4CF
__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
__CFRunLoopRun
CFRunLoopRunSpecific
GSEventRunModal
-[UIApplication _run]
UIApplicationMain
main
start
```

On iOS, datepicker violently crashes when users select a non-gregorian calendar

**Expected behavior**

datepicker shouldn't crash / can be set to display specified calendar

**To Reproduce**

1. Set device calendar to Japanese calendar (iOS settings > Language & Region > Calendar > Japanese)
2. Run RN app

```javascript
export default class App extends Component {

  state = { date: new Date() }

  render = () =>
    <DatePicker
      date={this.state.date}
      locale='en-SG'
      onDateChange={date => this.setState({ date })}
    />

}
```

**Smartphone (please complete the following information):**
 - OS: iOS 14.4
 - React Native version 62.4
 - react-native-date-picker version 3.2.10

**Comment by sbycrosz** [comment]
For our use case, we ended up hardcoding the attribute to gregorian calendar on init

**patches/react-native-date-picker+3.2.10.patch**
```
diff --git a/node_modules/react-native-date-picker/ios/RNDatePicker/DatePicker.m b/node_modules/react-native-date-picker/ios/RNDatePicker/DatePicker.m
index a37dc42..2de02f4 100644
--- a/node_modules/react-native-date-picker/ios/RNDatePicker/DatePicker.m
+++ b/node_modules/react-native-date-picker/ios/RNDatePicker/DatePicker.m
@@ -58,6 +58,11 @@ - (instancetype)initWithFrame:(CGRect)frame
             self.preferredDatePickerStyle = UIDatePickerStyleWheels;
         }
          _reactMinuteInterval = 1;
+
+        // Always use gregorian calendar regardless of user's preferred calendar
+        // TODO: Update this library once the issue is fixed
+        //       https://github.com/henninghall/react-native-date-picker/issues/344
+        self.calendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierGregorian];
     }
     return self;
 }
@@ -133,4 +138,3 @@ - (void)setMinuteInterval:(NSInteger)minuteInterval
 
 
 @end
-
```

But it would probably more appropriate to be able to set this attribute from JavaScript ✌️

**Comment by henninghall** [comment]
Released a fix for this in version `3.4.0`
